src/guest_memory.rs

```diff
@@ -586,6 +587,168 @@ pub trait GuestMemory {
     }
 }
 
+/// Errors that can occur when creating a memory map.
+#[derive(Debug, thiserror::Error)]
+pub enum GuestRegionCollectionError {
```

Comment:
This whole `GuestRegionCollection` looks like it can be moved into it's own module. Also maybe rename to just `Regions`? I assume there are no other regions other than memory regions.

Reply:
Mh, I'd be open to having it in its own module. What should we call it though?

Reply:
Just `regions.rs`?

Reply:
done